### ISSUE TITLE
Support jvp for flex attention

### ISSUE THREAD

**Issue by Birch-san** [pytorch/pytorch#154332]
### 🚀 The feature, motivation and pitch

A derivation for the jvp of attention was shared in [Simplifying, Stabilizing and Scaling Continuous-Time Consistency Models](https://arxiv.org/abs/2410.11081) (Cheng Lu, Yang Song (OpenAI)). Significantly, it can be computed blockwise (i.e. QK similarities fused with online softmax).

It would be great if flex attention could implement this, to support forward-mode autodiff!

There's a lot of research interest in few-step diffusion/flow models, e.g. for image generation (and [text generation](https://deepmind.google/models/gemini-diffusion/) now too I suppose). Some highlights that utilize jvp as part of their approach:

- [NVIDIA Genie](https://research.nvidia.com/labs/toronto-ai/GENIE/)
- [Consistency models](https://proceedings.mlr.press/v202/song23a.html)
  - certainly jvp is used in consistency distillation, I'm less sure whether it's used in consistency training, where [iCT](https://arxiv.org/abs/2310.14189) is the more recent development
- [TrigFlow models](https://arxiv.org/abs/2410.11081)
  - this has been further incorporated into [Unified Continuous Generative Models](https://arxiv.org/abs/2505.07447), unclear whether they use jvp though
- [MeanFlow](https://arxiv.org/abs/2505.13447)
  - practitioners are already [implementing this](https://x.com/LodestoneE621/status/1926520450068124037) using jvp approximations (finite difference method)
- inferencing diffusion models via [Taylor method sampler](https://discord.com/channels/729741769192767510/1007710893024481380/1140763177508945950)
  - without retraining, you can get the benefits of NVIDIA Genie, using forward-mode autodiff to obtain the second derivative of the reverse diffusion ODE, to more accurately traverse curved ODE trajectories

#### The derivation

From Appendix F, https://arxiv.org/abs/2410.11081:

<img width="688" alt="Image" src="https://github.com/user-attachments/assets/5eaa49ce-968c-4e3b-9d51-4328a4bde9db" />

<img width="713" alt="Image" src="https://github.com/user-attachments/assets/b1ea26b1-7d99-4af5-8595-53505014e9b8" />

**Comment by Birch-san** [comment]
btw, @Ryu1845 vibe-coded a triton chunked attn jvp which seems to satisfy allclose; might be a useful reference.  
https://github.com/Dao-AILab/flash-attention/issues/1672#issuecomment-2927974809
